From 1523fc2795cd7159c084fdb2f6a68f46fa2ab4d0 Mon Sep 17 00:00:00 2001
From: Alexandre Bailon <abailon@baylibre.com>
Date: Mon, 14 Jun 2021 11:27:05 +0200
Subject: [PATCH 4/4] Add an Android.bp to build APU drm

This adds an Android.bp, required to build the libdrm
for the APU.

Signed-off-by: Alexandre Bailon <abailon@baylibre.com>
Change-Id: I968b81bafeae00a43342d349eb77670469d8a41f
---
 Android.bp             |  2 +-
 apu/Android.bp         | 12 ++++++++++++
 apu/Android.sources.bp |  9 +++++++++
 apu/meson.build        | 10 ----------
 4 files changed, 22 insertions(+), 11 deletions(-)
 create mode 100644 apu/Android.bp
 create mode 100644 apu/Android.sources.bp

diff --git a/Android.bp b/Android.bp
index 158f2660..8fbb4434 100644
--- a/Android.bp
+++ b/Android.bp
@@ -58,7 +58,7 @@ cc_library {
         "libdrm_sources",
     ],
 
-    export_include_dirs: ["include/drm", "android"],
+    export_include_dirs: ["include/drm", "android", "."],
 
     cflags: [
         "-Wno-enum-conversion",
diff --git a/apu/Android.bp b/apu/Android.bp
new file mode 100644
index 00000000..d4cd4c3a
--- /dev/null
+++ b/apu/Android.bp
@@ -0,0 +1,12 @@
+build = ["Android.sources.bp"]
+
+cc_library_shared {
+    name: "libdrm_apu",
+    defaults: [
+        "libdrm_defaults",
+        "libdrm_apu_sources",
+    ],
+    vendor: true,
+
+    shared_libs: ["libdrm"],
+}
diff --git a/apu/Android.sources.bp b/apu/Android.sources.bp
new file mode 100644
index 00000000..710870eb
--- /dev/null
+++ b/apu/Android.sources.bp
@@ -0,0 +1,9 @@
+// Autogenerated with Android.sources.bp.mk
+
+cc_defaults {
+    name: "libdrm_apu_sources",
+    srcs: [
+	"apu_drm.c",
+    ],
+    export_include_dirs: ["."],
+}
diff --git a/apu/meson.build b/apu/meson.build
index 794598a9..29e0da8a 100644
--- a/apu/meson.build
+++ b/apu/meson.build
@@ -45,13 +45,3 @@ pkg.generate(
   requires_private : 'libdrm',
   description : 'Userspace interface to apu kernel DRM services',
 )
-
-test(
-  'apu-symbols-check',
-  symbols_check,
-  args : [
-    '--lib', libdrm_apu,
-    '--symbols-file', files('apu-symbols.txt'),
-    '--nm', prog_nm.path(),
-  ],
-)
-- 
2.33.0

